libpromises/evalfunction.c

```diff
@@ -6888,6 +6889,47 @@ static int BuildLineArray(EvalContext *ctx, const Bundle *bundle,
 
 /*********************************************************************/
 
+static FnCallResult FnCallProcessExists(ARG_UNUSED EvalContext *ctx, ARG_UNUSED const Policy *policy, const FnCall *fp, const Rlist *finalargs)
+{
+    char *regex = RlistScalarValue(finalargs);
+
+    const bool context_mode = strcmp(fp->name, "processexists") == 0;
+
+    Item* procdata = NULL;
+    if (!LoadProcessTable(&procdata))
+    {
+        Log(LOG_LEVEL_ERR, "%s: could not load the process table?!?!", fp->name);
+        return FnFailure();
+    }
+
+    ProcessSelect ps;
+    Item *matched = SelectProcesses(procdata, regex, ps, false);
+    DeleteItemList(procdata);
+
+    if (context_mode)
+    {
+        return FnReturnContext(NULL != matched);
```

Comment:
We are having memory leak here. ```matched``` is allocated by ```SelectProcesses()``` but is never freed. Maybe something like ```bool ret = NULL != matched; DeleteItemList(matched); return ret;``` could be used instead.